### PR TITLE
Check balance(preState) = balance(postState) in testeth

### PR DESCRIPTION
**Pull request by winsvega**

### CODE REVIEW

File: test/tools/jsontests/BlockChainTests.cpp [hunk 1]
```diff
@@ -431,15 +434,20 @@ void testBCTest(json_spirit::mObject& _o)
 		}
 
 		// ImportUncles
+		vector<u256> uncleNumbers;
 		if (blObj["uncleHeaders"].type() != json_spirit::null_type)
+		{
+			BOOST_REQUIRE_MESSAGE(blObj["uncleHeaders"].get_array().size() <= 2, "Too many uncle headers in block! " + TestOutputHelper::testName());
```
Comment (pirapira) [inline test/tools/jsontests/BlockChainTests.cpp]: So, 0, 1 and 2 are OK but 3 is not OK?

File: test/tools/jsontests/BlockChainTests.cpp [hunk 2]
```diff
@@ -388,6 +389,8 @@ void testBCTest(json_spirit::mObject& _o)
 	{
 		mObject blObj = bl.get_obj();
 		TestBlock blockFromRlp;
+		State preState = testChain.topBlock().state();
```
Comment (pirapira) [inline test/tools/jsontests/BlockChainTests.cpp]: You can make this `const`.

File: test/tools/jsontests/BlockChainTests.cpp [hunk 3]
```diff
@@ -388,6 +389,8 @@ void testBCTest(json_spirit::mObject& _o)
...
+		h256 preHash = testChain.topBlock().blockHeader().hash();
```
Comment (pirapira) [inline test/tools/jsontests/BlockChainTests.cpp]: You can also make this `const`.

File: test/tools/jsontests/BlockChainTests.cpp (hunk 1 above)
Comment (winsvega) [inline test/tools/jsontests/BlockChainTests.cpp]: yes. block should have no more then 2 uncles

File: test/tools/libtesteth/ImportTest.cpp [hunk 4]
```diff
@@ -201,6 +201,19 @@ bytes ImportTest::executeTest()
...
+{
+	bigint preBalance = 0;
+	bigint postBalance = 0;
+	for (auto const& addr : _pre.addresses())
+		preBalance += addr.second;
+	for (auto const& addr : _post.addresses())
+		postBalance += addr.second;
+
+	//account could destroy ether if it suicides to itself
+	BOOST_REQUIRE_MESSAGE(preBalance + _miningReward >= postBalance, "Error when comparing states: preBalance != postBalance (" + toString(preBalance) + " != " + toString(postBalance) + ") " + TestOutputHelper::testName());
```
Comment (pirapira) [inline test/tools/libtesteth/ImportTest.cpp]: The error message should say `preBalance + miningReward < postBalance`.

File: test/tools/jsontests/BlockChainTests.cpp (hunk 1 above)
Comment (pirapira) [inline test/tools/jsontests/BlockChainTests.cpp]: OK.

**Pull request merged**